### PR TITLE
Release 0.0.1

### PR DESCRIPTION
**Pull request by kodesiba**
See [Release Notes](https://github.com/atorus-research/timber/releases/tag/v0.0.1) for information

### CODE REVIEW

**Comment by bms63** [comment]
Some of the lines are going over 80 characters?  Happens in a few other places

image

**Comment by bms63** [comment]
I think this should start with a Capital E? 
image

**Comment by bms63** [comment]
What do the asterisk mean?  Actively used in the script? Thought it was cool that the log capture that the timber package was loaded locally. 

image

**Comment by bms63** [comment]
Really want to change the base colors of the addin!!

**Comment by kodesiba** [comment]
@bms63 the lowercase errors should have been fixed, I'm not seeing it on my local build from dev.  I also added some extra pieces to make sure the session info section doesn't end up over 80 characters, good catch on that

File: R/zzz.R [hunk 1]
```diff
@@ -1,5 +1,14 @@
...
+   # init functions for other timber functionality
    log_init()
+   approved_functions_init()
+
+   # This overwrite is to correctly build purrr adverb function
+   # outlined in purrr best practices for exporting adverb-wrapped functions
+   run_safely <<- purrr::safely(run_file, quiet = FALSE)
+
+   # set warn to 1 to have warnings be output as they happen
+   options(warn = 1)
```
Comment (nicholas-masel) [inline R/zzz.R:13]: Do we need to alert the user we are changing their options?

**Comment by bms63** [comment]
image

File: tests/testthat/ref/ex1.log [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+timber link to repository: https://github.com/atorus-research/timber
+--------------------------------------------------------------------------------
+-                               User Information                               -
+--------------------------------------------------------------------------------
+User: nmasel
+--------------------------------------------------------------------------------
+-                             Session Information                              -
+--------------------------------------------------------------------------------
+─ Session info  ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
+ hash: family: man, man, girl, girl, rolled-up newspaper, Ophiuchus
```
Comment (nicholas-masel) [inline tests/testthat/ref/ex1.log:16]: The random emoji hash is removed in the latest version of [sessioninfo](https://github.com/r-lib/sessioninfo/commit/1f38eeaba6fdbf292340199b443b60d2613d0409).  Do we want to upgrate sessioninfo and rerun these examples to update the log?  I'm ok either way.

**Review by nicholas-masel — COMMENTED** [review]
Minor comments for consideration.

File: R/get.R [hunk 3]
```diff
@@ -82,12 +82,20 @@ get_file_path <- function(file = NA, normalize = TRUE){
 #' @export
 #'
 #' @importFrom sessioninfo session_info
+#' @importFrom purrr map
+#' @importFrom stringr str_wrap
```
Comment (bms63) [inline R/get.R]: I think `str_wrap` needs to get declared in exports/NAMESPACE.

File: R/get.R [hunk 4]
```diff
@@ -82,12 +82,20 @@ get_file_path <- function(file = NA, normalize = TRUE){
...
 #'
 #' @examples
 #' get_session_info()
 #'
 get_session_info <- function(){
```
Comment (bms63) [inline R/get.R:89]: This update might cause some formatting issues after the wrap.  I noticed the .log was slightly different between the two times I have run .  
image

**Comment by bms63** [comment]
> @bms63 the lowercase errors should have been fixed, I'm not seeing it on my local build from dev. I also added some extra pieces to make sure the session info section doesn't end up over 80 characters, good catch on that

Cool I was in wrong branch when I loaded the package.  Errors is there.  Thanks!

**Comment by kodesiba** [comment]
@bms63 @nicholas-masel checks are finally passing again, I did some stuff to make the lines 80 characters which I think will work for now.  If y'all have anything else let me know

**Comment by bms63** [comment]
Wrapping and formatting is looking almost perfect - I noticed this goes over the line now

image

**Comment by bms63** [comment]
I just noticed the website is getting 404ed.  You guys seeing this?  https://atorus-research.github.io/timber/.  I looked in the workflows and maybe i did something with a PR?  Not sure if I need to be an admin to fix?

**Comment by kodesiba** [comment]
> I just noticed the website is getting 404ed. You guys seeing this? https://atorus-research.github.io/timber/. I looked in the workflows and maybe i did something with a PR? Not sure if I need to be an admin to fix?

it's down because the vignette was broken, on the merge to main the site will rebuild

**Comment by kodesiba** [comment]
> 

I updated this in the errors writer.  There is also now an updated function in the session info writer that uses stri_wrap that can do wrapping of any element though I hesitate to run everything through it currently without taking some time to test it more.  It's just a slightly changed str_wrap but without the whitespace collapsing

**Comment by bms63** [comment]
I was running the admiral::use_ad_template("adsl") to make an adsl.  The Results section of the log file gets filled out with the final dataset.   Is that intentional.  

image

This is way cool!! 
image


Should things like this get captured? Or is this a `lifecycle `related warning so doesn't get capture in log?
image

**Comment by kodesiba** [comment]
@bms63 responding here
1- yes that is intentional, we might update in the future to have the user specify what pieces of messages/output/result they want to log for now but if the script returns a result it'll get logged

2- that's awesome, I love seeing that functionality in action and can't believe that @nicholas-masel was able to get that in

3- I think that's not there because it is lifecycle, we can look at that after this release as a possible thing we need to address

**Pull request merged**